tests/verify-strings/test_scrypt.c

```diff
@@ -24,16 +24,24 @@ main(int argc, char * argv[])
 	struct scrypt_test * test;
 	char kbuf[64];
 	size_t i;
+	size_t num_tests = sizeof(tests) / sizeof(tests[0]);
+	size_t test_i;
 	int failures = 0;
+	int skip_largemem = 0;
 
 	WARNP_INIT;
 
-	(void)argc; /* UNUSED */
-	(void)argv; /* UNUSED */
+	/* Do we want to skip the large-memory test? */
+	if ((argc == 2) && (strcmp(argv[1], "1") == 0))
+		skip_largemem = 1;
+
+	for (test_i = 0; test_i < num_tests; test_i++) {
+		test = &tests[test_i];
+
+		/* Skip the 1 GB memory test (if desired). */
+		if ((skip_largemem) && (test_i == 3))
```

Comment:
I would have written this as `test == &tests[3]` and avoided the whole `test_i` thing...?

Reply:
Alternatively, added an `int largemem;` field to `struct scrypt_test` and then this could just be `if (skip_largemem && test->largemem)`.

Reply:
Do you really like the original way of looping?  It throws me for a loop (pun intended) whenever I look at that file.

I have to admit that I like the `int largemem` solution, though.

Reply:
I'm not opposed to the original way of looping.  Not sure if I would write it like that now, but I wouldn't change it without a good reason either.